test/services/DbPatchServiceSpec.scala

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import config.AppConfig
+import utils.SpecBase
+
+import scala.concurrent.{ExecutionContext, Future}
+import org.mongodb.scala._
+import org.mongodb.scala.result.DeleteResult
+import play.api.test.Helpers.running
+import play.api.{Application, inject}
+import uk.gov.hmrc.mongo.MongoComponent
+import utils.Utils.emptyString
+
+class DbPatchServiceSpec extends SpecBase {
+
+  "Database Patch Service" should {
+
+    "delete documents when deleteGuaranteeAccountCacheDocuments is true" in new Setup {
```

Comment:
Better to update the test description to as below as feature flag info should be highlighted as that is the main bit here. Cause, deleteGuaranteeAccountCacheDocuments without any reference to config  is not adding that much value in the description. same for test starts at line 47

"delete documents when 'delete-guarantee-account-cache-documents' feature flag is true"

or "delete documents when config.deleteGuaranteeAccountCacheDocuments is true"

Reply:
Renamed as per comment